src/libraries/Common/tests/System/IO/ByteLoggingStream.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO
+{
+    internal sealed class BytesLoggingStream : Stream
+    {
+        private readonly Stream _stream;
+        private readonly Action<bool, string> _log;
+
+        public BytesLoggingStream(Stream stream, Action<bool, string> log)
+        {
```

Comment:
This is very much a nit, but I wonder if you should have two different actions for logRead and logWrite. And we might add logClose in the future, or even logSeek etc.